### DISCUSSION TITLE
Is there a concept to localize the names of sensors?

### DISCUSSION THREAD

**Discussion by klacol** [home-assistant/architecture#855] [General] [answered]
Hi, in my small integration for a continuous waterheater, I have defined my sensors in this [file ](https://github.com/klacol/homeassistant-clage_homeserver/blob/master/custom_components/clage_homeserver/sensor.py) based on a meta class that defines the structure of a sensor.

It is like this:

```python
_sensors = [
    SensorDefinition(
        system_name="homeserver_version",
        name="Homeserver Version",
        definition="Version der Software und des API des Homeservers",
        unit=None,
        state_class=STATE_CLASS_MEASUREMENT,
        device_class=None,
        entity_category=EntityCategory.DIAGNOSTIC,
    ),
...
```

Does anyone know, If there is a concept, how I can provide the names and definitions of the sensors in a pre localizes way, so that the users do not have to translate/rename them manually after installation?

**Comment by frenck** [comment, marked as answer]
There is not the place for development support, please join our Discord server and use the #dev_* channels for development questions instead.

To answer your question: There is no such concept as this point. The recent additions to our translation do prepare for such features, but we have yet to arrive / we have yet to build.

../Frenck

**Reply by klacol** [reply]
Thanks for this information.

**Comment by klacol** [comment]
This seems to address my point a bit, mostly related to state names:

https://www.home-assistant.io/blog/2023/01/04/release-20231/#translation-improvements-for-entities

https://github.com/home-assistant/core/blob/dev/homeassistant/components/moon/translations/en.json#L14

I will have a look, what the rule will be in the future, to add localized sensor names.
